pkg/webhooks/leaderworkerset_webhook.go

```diff
@@ -165,7 +165,7 @@ func (r *LeaderWorkerSetWebhook) generalValidate(obj runtime.Object) field.Error
 	if err != nil {
 		allErrs = append(allErrs, field.Invalid(maxSurgePath, maxSurge, "invalid value"))
 	}
-	if maxUnavailableValue == 0 && maxSurgeValue == 0 {
+	if maxUnavailableValue == 0 && maxSurgeValue == 0 && *lws.Spec.Replicas != 0 {
```

Comment:
I think it's not related between the scaling-to-zero and rolling update configuration. What you mean here is once replicas == 0, the maxUnavailable and maxSurgeValue could both be zero, however, it's not.

Reply:
https://github.com/kubernetes-sigs/lws/blob/da1267b252e17fe795cbc41249a43eb9d52cbca5/pkg/webhooks/leaderworkerset_webhook.go#L160C37-L160C67

Let's consider the case: replica = 0, maxUnavailable = 25% and maxSurge = 20%.
Then maxUnavailableValue will be 0 and maxSurge will be 0 as the calculated as above. Then admission webhook forbid this operation.

Reply:
Sorry, I forgot the value is calculated from the replicas. Let's add a webhook test to make sure it works with replicas==0.